ext/node/ops/ipc.rs

```diff
@@ -43,6 +53,106 @@ mod impl_ {
   #[cfg(windows)]
   type NamedPipeClient = tokio::net::windows::named_pipe::NamedPipeClient;
 
+  /// Wrapper around v8 value that implements Serialize.
+  struct SerializeWrapper<'a, 'b>(
+    RefCell<&'b mut v8::HandleScope<'a>>,
+    v8::Local<'a, v8::Value>,
+  );
+
+  impl<'a, 'b> Serialize for SerializeWrapper<'a, 'b> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+      S: Serializer,
+    {
+      serialize_v8_value(*self.0.borrow_mut(), self.1, serializer)
+    }
+  }
+
+  /// Serialize a v8 value directly into a serde serializer.
+  /// This allows us to go from v8 values to JSON without having to
+  /// deserialize into a `serde_json::Value` and then reserialize to JSON
+  fn serialize_v8_value<'a, S: Serializer>(
```

Comment:
Could we get a few unit test cases? This is quite complex and is prone to changes in `rusty_v8` APIs.

Reply:
Yeah I added a test to `child_process_test.ts` to exercise this, but I'll add some here too that test it more directly.